src/mappings/entities/mapStatistics.ts

```diff
@@ -256,11 +268,143 @@ const handleExemptionsRemoved = async (params: Codec[]) => {
   );
 };
 
+const handleTransferManagerAdded = async (blockId: string, params: Codec[]) => {
+  const [, rawTicker, rawManager] = params;
+
+  const ticker = serializeTicker(rawTicker);
+  const { type } = getTransferManagerValue(rawManager);
+
+  if (type === TransferRestrictionTypeEnum.Percentage) {
+    const statId = `${ticker}/${StatOpTypeEnum.Balance}`;
+    const stat = await StatType.get(statId);
+    if (!stat) {
+      await StatType.create({
+        id: statId,
+        opType: StatOpTypeEnum.Balance,
+        assetId: ticker,
+        claimType: null,
+        claimIssuerId: null,
+        createdBlockId: blockId,
+        updatedBlockId: blockId,
+      }).save();
+    }
+  }
```

Comment:
You could refactor this as 
```
const createStatType = async (
  { assetId, opType, claimType, claimIssuerId }: Attributes<StatType>,
  blockId: string
) => {
  const statTypeId = getStatTypeId(assetId, opType, claimType, claimIssuerId);
  let statType = await StatType.get(statTypeId);
  if (statType) {
    statType.updatedBlockId = blockId;
  } else {
    statType = StatType.create({
      id: statTypeId,
      assetId,
      opType,
      claimType,
      claimIssuerId,
      createdBlockId: blockId,
      updatedBlockId: blockId,
    });
  }
  return statType.save();
};
```

And call this method wherever StatType.create is invoked. Similarly other code refactoring can be done so that, we have single point from where StatType.create can be called